### ISSUE TITLE
Uncaught TypeError: Cannot read property 'dispose' of undefined

### ISSUE THREAD

**Issue by DrDanRyan** [atom/tabs#183]
[Enter steps to reproduce below:]

1. Open Git-Log: Show
2. Try to close tab

**Atom Version**: 0.210.0
**System**: Mac OS X 10.10.3
**Thrown From**: [tabs](https://github.com/atom/tabs) package, v0.81.0

### Stack Trace

Uncaught TypeError: Cannot read property 'dispose' of undefined

```
At /Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:25

TypeError: Cannot read property 'dispose' of undefined
  at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:25:28
  at Set.forEach (native)
  at CompositeDisposable.module.exports.CompositeDisposable.dispose (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:24:26)
  at tabs-tab.module.exports.TabView.destroy (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-view.js:194:14)
  at TabBarView.module.exports.TabBarView.removeTabForItem (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:364:15)
  at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:174:24
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
  at Pane.module.exports.Pane.removeItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:379:20)
  at Pane.module.exports.Pane.destroyItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:442:16)
  at TabBarView.module.exports.TabBarView.closeTab (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:419:26)
  at Object.addElementCommands.tabs:close-tab (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:95:26)
  at space-pen-ul.commandsWithPropagationStopped.(anonymous function) (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:86:36)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:238:29)
  at CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61)
  at CommandRegistry.module.exports.CommandRegistry.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:153:19)
  at [object Object].jQuery.fn.trigger (/Applications/Atom.app/Contents/Resources/app.asar/src/space-pen-extensions.js:64:23)
  at EventEmitter.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:75:67)
  at emitTwo (events.js:87:13)
  at EventEmitter.emit (events.js:169:7)
  at EventEmitter.ipc.sendSync (/Applications/Atom.app/Contents/Resources/atom.asar/renderer/api/lib/ipc.js:21:31)
  at BrowserWindow.RemoteMemberFunction [as emit] (/Applications/Atom.app/Contents/Resources/atom.asar/renderer/api/lib/remote.js:110:29)
  at ContextMenuManager.module.exports.ContextMenuManager.showForEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/context-menu-manager.js:170:31)
  at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:150:33)
  at HTMLDocument.handler (/Applications/Atom.app/Contents/Resources/app.asar/src/space-pen-extensions.js:112:34)
  at HTMLDocument.jQuery.event.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/space-pen/vendor/jquery.js:4681:9)
  at HTMLDocument.elemData.handle (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/space-pen/vendor/jquery.js:4359:46)

```

### Commands

```
     -4:55.8.0 pane:close (atom-text-editor.editor.is-focused)
     -4:38.9.0 fuzzy-finder:toggle-file-finder (atom-text-editor.editor.autocomplete-active.is-focused)
     -4:37.4.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  6x -3:36.2.0 core:backspace (atom-text-editor.editor.is-focused)
     -3:16.8.0 core:delete (atom-text-editor.editor.is-focused)
     -3:14 core:move-down (atom-text-editor.editor.is-focused)
     -3:13.3.0 editor:select-line (atom-text-editor.editor.is-focused)
     -3:13.1.0 editor:delete-to-beginning-of-line (atom-text-editor.editor.is-focused)
     -3:12.7.0 core:save (atom-text-editor.editor.is-focused)
     -0:27.4.0 command-palette:toggle (atom-text-editor.editor.is-focused)
     -0:24 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:24 git-log:show (atom-text-editor.editor)
     -0:20 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:12.7.0 core:close (div.git-log)
```

### Config

```json
{
  "core": {
    "themes": [
      "seti-ui",
      "saturated-dark"
    ],
    "projectHome": "/Users/mothership/Code",
    "destroyEmptyPanes": false,
    "disabledPackages": [
      "language-javascript-semantic",
      "tidy-markdown",
      "atom-beautify",
      "atom-typescript",
      "pigments"
    ],
    "audioBeep": false
  }
}
```

### Installed Packages

```coffee
# User
.bin, vundefined
angularjs, v0.3.0
color-picker, v2.0.5
emmet, v2.3.10
fonts, v0.4.2
git-log, v0.4.1
git-plus, v5.1.2
git-tab-status, v1.9.2
linter, v0.12.7
linter-eslint, v0.5.9
linter-jscs, v1.12.0
minimap, v4.10.0
minimap-find-and-replace, v4.2.0
minimap-git-diff, v4.1.3
minimap-selection, v4.2.0
project-manager, v1.15.10
saturated-dark, v0.4.0
seti-ui, v0.7.1
zentabs, v0.8.4

# Dev
No dev packages
```

**Comment by Fank** [comment]
I got the same error, with the same package. Open the git log, close it before it show content and close it.

**Atom Version**: 0.210.0
**System**: Ubuntu 15.04
**Thrown From**: [tabs](https://github.com/atom/tabs) package, v0.81.0


### Stack Trace

Uncaught TypeError: Cannot read property 'dispose' of undefined

```
At /usr/share/atom/resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:25

TypeError: Cannot read property 'dispose' of undefined
  at /usr/share/atom/resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:25:28
  at Set.forEach (native)
  at CompositeDisposable.module.exports.CompositeDisposable.dispose (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:24:26)
  at tabs-tab.module.exports.TabView.destroy (/usr/share/atom/resources/app.asar/node_modules/tabs/lib/tab-view.js:194:14)
  at TabBarView.module.exports.TabBarView.removeTabForItem (/usr/share/atom/resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:364:15)
  at /usr/share/atom/resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:174:24
  at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
  at Pane.module.exports.Pane.removeItem (/usr/share/atom/resources/app.asar/src/pane.js:379:20)
  at Pane.module.exports.Pane.destroyItem (/usr/share/atom/resources/app.asar/src/pane.js:442:16)
  at HTMLDivElement.<anonymous> (/usr/share/atom/resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:236:22)
  at space-pen-ul.jQuery.event.dispatch (/usr/share/atom/resources/app.asar/node_modules/jquery/dist/jquery.js:4435:9)
  at space-pen-ul.elemData.handle (/usr/share/atom/resources/app.asar/node_modules/jquery/dist/jquery.js:4121:28)

```

### Commands

```
     -1:32.6.0 command-palette:toggle (atom-text-editor.editor.is-focused)
     -1:30.1.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -1:30.1.0 git-log:show (atom-text-editor.editor)
     -1:28.7.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -1:28.2.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -1:06.3.0 command-palette:toggle (atom-text-editor.editor.is-focused)
     -1:02.8.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -1:02.4.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -1:02.3.0 git-plus:log (atom-text-editor.editor)
     -0:28.1.0 core:backspace (div.settings-view.pane-item)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "snippets",
      "linter-perl",
      "atom-auto-uglify",
      "save-session",
      "highlight-selected",
      "minimap-highlight-selected",
      "atom-2048",
      "achievements"
    ],
    "autoHideMenuBar": true
  }
}
```

### Installed Packages

```coffee
# User
autocomplete-paths, v1.0.2
color-picker, v2.0.5
default-language, v0.1.3
git-log, v0.4.1
git-plus, v5.1.2
git-tab-status, v1.9.2
html-entities, v0.4.0
language-mason, v2.0.4
language-perl, v0.25.0
linter, v0.12.7
linter-jshint, v0.1.6
linter-lua, v0.1.6
minimap, v4.10.0
minimap-find-and-replace, v4.2.0
minimap-git-diff, v4.1.3
minimap-selection, v4.2.0
set-syntax, v0.3.0
tabs-to-spaces, v0.11.0
travis-ci-status, v0.15.1

# Dev
No dev packages
```

**Comment by DjebbZ** [comment]
Same here, it happened to me just now.

**Comment by filipesilva** [comment]
I also have this problem.

**Comment by denysdovhan** [comment]
I have same problem

**Comment by damm** [comment]
same here...  Nothing fancy started atom closed 2 windows and ..

***Atom Version**: 0.211.0
**System**: Mac OS X 10.10.3
**Thrown From**: [tabs](https://github.com/atom/tabs) package, v0.81.0


### Stack Trace

Uncaught Error: The workspace can only contain one instance of item [object Object]

```
At /Applications/Atom.app/Contents/Resources/app.asar/src/item-registry.js:11

Error: The workspace can only contain one instance of item [object Object]
    at ItemRegistry.module.exports.ItemRegistry.addItem (/Applications/Atom.app/Contents/Resources/app.asar/src/item-registry.js:11:15)
    at PaneContainer.module.exports.PaneContainer.addedPaneItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane-container.js:374:25)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/pane-container.js:362:26
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
    at Pane.module.exports.Pane.addItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:326:20)
    at Pane.module.exports.Pane.activateItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:297:14)
    at tabs-tab.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:210:24)
    at space-pen-ul.jQuery.event.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4435:9)
    at space-pen-ul.elemData.handle (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4121:28)
```

### Commands

```
     -0:09.6.0 pane:show-next-item (atom-text-editor.editor.is-focused)
     -0:08.9.0 core:close (div.settings-view.pane-item)
     -0:07.6.0 pane:show-next-item (atom-text-editor.editor.is-focused)
     -0:06.8.0 core:close (atom-text-editor.editor.is-focused)
     -0:05.1.0 pane:show-item-2 (atom-text-editor.editor.is-focused)
     -0:04.5.0 core:close (atom-text-editor.editor.is-focused)
     -0:03.2.0 pane:show-next-item (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-one-dark-ui)
     -0:02.8.0 pane:show-previous-item (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-one-dark-ui)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "metrics",
      "sourcegraph-atom",
      "snippets",
      "autocomplete-atom-api"
    ]
  }
}
```

### Installed Packages

```coffee
# User
No installed packages

# Dev
No dev packages
```

**Comment by nrozic** [comment]
Same here
[Enter steps to reproduce below:]

1. Open git log
2. Try to close git log tab

**Atom Version**: 1.0.0
**System**: Mac OS X 10.10.3
**Thrown From**: [tabs](https://github.com/atom/tabs) package, v0.81.0


### Stack Trace

Uncaught TypeError: Cannot read property 'dispose' of undefined

```
At /Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:25

TypeError: Cannot read property 'dispose' of undefined
  at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:25:28
  at Set.forEach (native)
  at CompositeDisposable.module.exports.CompositeDisposable.dispose (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:24:26)
  at tabs-tab.module.exports.TabView.destroy (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-view.js:194:14)
  at TabBarView.module.exports.TabBarView.removeTabForItem (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:364:15)
  at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:174:24
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
  at Pane.module.exports.Pane.removeItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:379:20)
  at Pane.module.exports.Pane.destroyItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:442:16)
  at TabBarView.module.exports.TabBarView.closeTab (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:419:26)
  at Object.addElementCommands.tabs:close-tab (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:95:26)
  at space-pen-ul.commandsWithPropagationStopped.(anonymous function) (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:86:36)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:241:29)
  at CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61)
  at CommandRegistry.module.exports.CommandRegistry.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:156:19)
  at [object Object].jQuery.fn.trigger (/Applications/Atom.app/Contents/Resources/app.asar/src/space-pen-extensions.js:64:23)
  at EventEmitter.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:75:67)
  at emitTwo (events.js:87:13)
  at EventEmitter.emit (events.js:169:7)
  at EventEmitter.ipc.sendSync (/Applications/Atom.app/Contents/Resources/atom.asar/renderer/api/lib/ipc.js:21:31)
  at BrowserWindow.RemoteMemberFunction [as emit] (/Applications/Atom.app/Contents/Resources/atom.asar/renderer/api/lib/remote.js:110:29)
  at ContextMenuManager.module.exports.ContextMenuManager.showForEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/context-menu-manager.js:170:31)
  at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:150:33)
  at HTMLDocument.handler (/Applications/Atom.app/Contents/Resources/app.asar/src/space-pen-extensions.js:112:34)
  at HTMLDocument.jQuery.event.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/space-pen/vendor/jquery.js:4681:9)
  at HTMLDocument.elemData.handle (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/space-pen/vendor/jquery.js:4359:46)

```

### Commands

```
     -5:48.7.0 command-palette:toggle (ol.tree-view.full-menu.list-tree.has-collapsable-children.focusable-panel)
     -5:44.3.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -5:44.2.0 git-log:show (ol.tree-view.full-menu.list-tree.has-collapsable-children.focusable-panel)
     -5:40.7.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -5:40.3.0 core:move-up (atom-text-editor.editor.mini.is-focused)
     -5:39.2.0 core:confirm (atom-text-editor.editor.mini.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
atom-material-syntax, v0.1.1
atom-material-ui, v0.4.9
flex-tool-bar, v0.4.2
git-diff-details, v0.18.0
git-log, v0.4.1
minimap, v4.10.0
minimap-bookmarks, v0.1.0
minimap-color-highlight, v4.1.4
minimap-git-diff, v4.1.3
tool-bar, v0.1.8
tool-bar-main, v0.0.8
tree-view-git-status, v0.1.1
wakatime, v4.0.10

# Dev
No dev packages
```

**Comment by alexbuzzbee** [comment]
Same.

1. Open a Git project in Atom.
2. Do git-log:show from command palate.
3. Wait for content to appear.
4. Attempt to close tab.
5. Notice that tab cannot be closed at all, even though, when selected, it is empty.

**Atom Version**: 1.0.0
**System**: Mac OS X 10.10.3
**Thrown From**: [tabs](https://github.com/atom/tabs) package, v0.81.0


### Stack Trace

Uncaught TypeError: Cannot read property 'dispose' of undefined

```
At /Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:25

TypeError: Cannot read property 'dispose' of undefined
  at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:25:28
  at Set.forEach (native)
  at CompositeDisposable.module.exports.CompositeDisposable.dispose (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:24:26)
  at tabs-tab.module.exports.TabView.destroy (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-view.js:194:14)
  at TabBarView.module.exports.TabBarView.removeTabForItem (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:364:15)
  at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:174:24
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
  at Pane.module.exports.Pane.removeItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:379:20)
  at Pane.module.exports.Pane.destroyItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:442:16)
  at HTMLDivElement.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:236:22)
  at space-pen-ul.jQuery.event.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4435:9)
  at space-pen-ul.elemData.handle (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4121:28)

```

### Commands

```
  5x -2:58.9.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -2:56.9.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:28.1.0 command-palette:toggle (ul.list-inline.tab-bar.inset-panel)
     -0:25.1.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:25.1.0 git-log:show (ul.list-inline.tab-bar.inset-panel)
  2x -0:22.2.0 core:confirm (atom-text-editor.editor.mini.is-focused)
```

### Config

```json
{
  "core": {
    "ignoredNames": [
      ".DS_Store"
    ],
    "projectHome": "~/Documents/Projects/Git",
    "themes": [
      "one-light-ui",
      "one-light-syntax"
    ],
    "useReactEditor": true,
    "followSymlinks": true,
    "disabledPackages": [
      "Tern"
    ]
  },
  "tabs": {
    "enableVcsColoring": true
  }
}
```

### Installed Packages

```coffee
# User
autocomplete-modules, v1.0.0
autocomplete-ruby, v0.1.0
build, v0.38.0
color-picker, v2.0.7
command-logger, v0.23.0
file-type-icons, v0.7.0
git-log, v0.4.1
language-arm, v0.3.3
language-lua, v0.9.3
language-m68k, v0.2.0
language-pascal, v0.7.0
language-prolog, v0.10.0
language-x86, v0.1.0
linter, v1.0.6
linter-coffeelint, v0.3.2
linter-cpplint, v0.1.4
linter-flake8, v1.4.2
linter-jshint, v0.1.7
linter-luacheck, v0.5.0
linter-pylint, v0.2.2
linter-rubocop, v0.2.7
linter-xmllint, v0.0.6
minimap, v4.10.0
minimap-bookmarks, v0.1.0
minimap-find-and-replace, v4.2.0
minimap-git-diff, v4.1.3
minimap-linter, v0.1.0
minimap-selection, v4.2.0

# Dev
No dev packages
```

**Comment by crucialfelix** [comment]
I'm the author of the atom supercollider package. I get the same issue all the time. I would guess its because there isn't a real document in that pane. Its a REPL post window.

**Comment by damm** [comment]
+1 what @crucialfelix said; I was trying to close several tabs and it looked like it spawned a bunch of empty tabs in error (with no documents)

I had to just quit Atom and start it back up and it worked

**Comment by weiher** [comment]
Off-Topic: I really love the "This issue has already been reported. View Issue" Function in Atom Editor. You know that there is a bug, you know how it can be reproduce and as all of your users are developers, they can handle this type of information! 

On-Topic: Had the same problem, **window:reload** as suggested by @alexbuzzbee works for me.

**Comment by mradzinski** [comment]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.0.0
**System**: Unknown Windows Version
**Thrown From**: [tabs](https://github.com/atom/tabs) package, v0.81.0


### Stack Trace

Uncaught TypeError: Cannot read property 'dispose' of undefined

```
At C:\Users\Admin\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\event-kit\lib\composite-disposable.js:25

TypeError: Cannot read property 'dispose' of undefined
    at C:\Users\Admin\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\event-kit\lib\composite-disposable.js:25:28
    at Set.forEach (native)
    at CompositeDisposable.module.exports.CompositeDisposable.dispose (C:\Users\Admin\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\event-kit\lib\composite-disposable.js:24:26)
    at tabs-tab.module.exports.TabView.destroy (C:\Users\Admin\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\tabs\lib\tab-view.js:194:14)
    at TabBarView.module.exports.TabBarView.removeTabForItem (C:\Users\Admin\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\tabs\lib\tab-bar-view.js:364:15)
    at C:\Users\Admin\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\tabs\lib\tab-bar-view.js:174:24
    at Emitter.module.exports.Emitter.emit (C:\Users\Admin\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:82:11)
    at Pane.module.exports.Pane.removeItem (C:\Users\Admin\AppData\Local\atom\app-1.0.0\resources\app.asar\src\pane.js:379:20)
    at Pane.module.exports.Pane.destroyItem (C:\Users\Admin\AppData\Local\atom\app-1.0.0\resources\app.asar\src\pane.js:442:16)
    at HTMLDivElement.<anonymous> (C:\Users\Admin\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\tabs\lib\tab-bar-view.js:236:22)
```

### Commands

```
     -0:21 command-palette:toggle (atom-text-editor.editor.is-focused)
     -0:18.4.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:18.4.0 git-log:show (atom-text-editor.editor)
     -0:17.1.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:16.8.0 core:confirm (atom-text-editor.editor.mini.is-focused)
```

### Config

```json
{
  "core": {
    "audioBeep": false
  }
}
```

### Installed Packages

```coffee
# User
atom-grails, v0.1.0
color-picker, v2.0.7
git-log, v0.4.1
language-groovy, v0.4.0
pigments, v0.8.2

# Dev
No dev packages
```

**Comment by motjuste** [comment]
1. check git log for a project in C++ (Package: git-log)
2. close the git log tab with Cmd+w

**Atom Version**: 1.0.0
**System**: Mac OS X 10.10.3
**Thrown From**: [tabs](https://github.com/atom/tabs) package, v0.81.0


### Stack Trace

Uncaught TypeError: Cannot read property 'dispose' of undefined

```
At /Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:25

TypeError: Cannot read property 'dispose' of undefined
  at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:25:28
  at Set.forEach (native)
  at CompositeDisposable.module.exports.CompositeDisposable.dispose (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/composite-disposable.js:24:26)
  at tabs-tab.module.exports.TabView.destroy (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-view.js:194:14)
  at TabBarView.module.exports.TabBarView.removeTabForItem (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:364:15)
  at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/tabs/lib/tab-bar-view.js:174:24
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
  at Pane.module.exports.Pane.removeItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:379:20)
  at Pane.module.exports.Pane.destroyItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:442:16)
  at Pane.module.exports.Pane.destroyActiveItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:419:12)
  at Workspace.module.exports.Workspace.destroyActivePaneItem (/Applications/Atom.app/Contents/Resources/app.asar/src/workspace.js:600:35)
  at Workspace.module.exports.Workspace.destroyActivePaneItemOrEmptyPane (/Applications/Atom.app/Contents/Resources/app.asar/src/workspace.js:639:21)
  at atom-workspace.atom.commands.add.core:close (/Applications/Atom.app/Contents/Resources/app.asar/src/workspace-element.js:308:30)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:241:29)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:520:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:355:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:180:20)

```

### Commands

```
  7x -0:39.4.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -0:28.1.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:27.2.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:27.2.0 git-plus:log (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-one-dark-ui)
     -0:23.9.0 core:close (div.git-plus-log.native-key-bindings)
     -0:23 command-palette:toggle (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-one-dark-ui)
     -0:18.9.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -0:17.9.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:17.9.0 git-plus:log (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-one-dark-ui)
     -0:15 core:close (div.git-plus-log.native-key-bindings)
     -0:14.2.0 command-palette:toggle (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-one-dark-ui)
     -0:11.7.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -0:09.7.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:09.7.0 git-log:show (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-one-dark-ui)
     -0:08.1.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:01.1.0 core:close (div.git-log)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "autocomplete-jedi",
      "clipboard-plus",
      "color-picker",
      "vim-mode-clipboard-plus",
      "linter-cpplint",
      "linter",
      "linter-pep8"
    ],
    "destroyEmptyPanes": false
  }
}
```

### Installed Packages

```coffee
# User
git-log, v0.4.1
git-plus, v5.2.2
highlight-selected, v0.10.1
language-matlab, v0.1.1
language-swift, v0.4.0
minimap, v4.10.1
set-syntax, v0.3.0
vim-mode, v0.54.0

# Dev
No dev packages
```

**Comment by as-cii** [comment]
This seems to be the culprit for `git-log-view`: https://github.com/NikhilKalige/git-log/blob/master/lib/git-log-view.js#L183-L189. I think those two functions were implemented because at some point in `tabs` history they were mandatory: the package, however, is not actually using them and unfortunately returns `null`. As a result, we are adding an `undefined` object to the `CompositeDisposable` that, of course, hasn’t got a `dispose` function. To solve this issue I am going to:

1. Put a guard in `event-kit`, allowing to add to a `CompositeDisposable` only those objects that respond to `dispose`. This should allow package authors to catch these kinds of issues early;
2. Open a PR in `git-log-view` and delete those two methods, as they are not mandatory to respect `TabView` contract;

/cc: @benogle @kevinsawicki

**Comment by thomasjo** [comment]
Nice catch @as-cii

**Comment by ManasN** [comment]
Can I overcome this issue temporarily and still use the plugin? Won't mind doing some extra settings or pressing extra keys.
I love atom. Slickest at it's best.

**Comment by as-cii** [comment]
@ManasN: there's a [pending PR](https://github.com/atom/event-kit/issues/12) where we have discussed how to properly address this problem. We have settled on adding a check to verify that clients properly implement the `Disposable` contract and, once it lands, you should be able to use the `git-log` package (even if https://github.com/NikhilKalige/git-log/pull/60 doesn't get merged).

Sorry this is taking a while to get fixed, but I'll try to make time this week to push the resolution forward. Thanks for understanding! :pray:

**Comment by gsscoder** [comment]
TL;DR
I also had this issue...

**Comment by filipesilva** [comment]
Cheers for the fix!

**Comment by gsscoder** [comment]
:+1:
